batch/batch/driver/instance_collection/job_private.py

```diff
@@ -179,12 +179,13 @@ async def schedule_jobs_loop_body(self):
         async for record in self.db.select_and_fetchall(
             '''
 SELECT jobs.*, batches.format_version, batches.userdata, batches.user, attempts.instance_name, time_ready
-FROM batches
-INNER JOIN jobs ON batches.id = jobs.batch_id
+FROM job_groups
+LEFT JOIN batches ON batches.id = job_groups.batch_id
```

Comment:
I usually try and use left joins rather than inner joins because the errors will be obvious if a row is not present when we expect it to be (NULL values will cause downstream errors). With an inner join, we could silently skip rows. But we do have all of the foreign key constraints so you're right. Let me know if you have a preference. Mine is not particularly strong

Reply:
> I usually try and use left joins rather than inner joins because the errors will be obvious if a row is not present when we expect it to be (NULL values will cause downstream errors)

This sounds very reasonable! I certainly errors and alerts to silent failures, can just keep this as is